### PR TITLE
Old Table: Table is not rendering when col property is undefined 

### LINKED ISSUE #44127: Old Table: Table is not rendering when col property is undefined
This issue was created as a follow up of this [escalation](https://github.com/grafana/support-escalations/issues/1650)

Old table panel breaks after saving in Grafana 8.3.0  

The root cause is related to this [PR](https://github.com/grafana/grafana/pull/41385) , in older versions, we have the sort property for panel.json like this:

Older versions ([Grafana play link](https://play.grafana.org/d/0kvGeE1nzsss/working-old-table-panel-not-rendering-in-new-versions-imported-grafana-6-5?orgId=1&inspect=2&inspectTab=json)): 
```
"sort": {
    "col": null,
    "desc": false
  },
```

but because that PR removes props that are `null`,  `col` prop disappear 🪄 

Here the new version after dashboard is modified [Error in Grafana play link](https://play.grafana.org/d/0kvGeE1nz/error-old-table-panel-not-rendering-in-new-versions-imported-grafana-6-5?orgId=1&inspect=2&inspectTab=json)
```
"sort": {
    "desc": false
  },
```
And the old table panel does not consider that edge case

### PR DESCRIPTION
**Pull request by axelavargas**
**What this PR does / why we need it**:

The old table panel does not support when `col` is undefined in the `sort` object (It does not render), This issue is present after `8.3.0` when this prop is null (null objects are stripped out, [PR that added that change](https://github.com/grafana/grafana/pull/41385)), I added an extra condition for that in the table model.

**Which issue(s) this PR fixes**:

Fixes #44127

**Special notes for your reviewer**:

I am not sure about the implications of doing this change in the model.

### CODE REVIEW

**Review by joshhunt — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
another bug from stripping nulls out eh... 😢 

looks good to me 🚀

**Review by hugohaggmark — APPROVED** [review]
LGTM! Any tests that need updating?

File: public/app/core/table_model.ts [hunk 1]
```diff
@@ -41,7 +41,8 @@ export default class TableModel implements TableData {
   }
 
   sort(options: { col: number; desc: boolean }) {
-    if (options.col === null || this.columns.length <= options.col) {
+    // Since 8.3.0 col property can be also undefined
+    if (options.col === null || options.col === undefined || this.columns.length <= options.col) {
```
Comment (ryantxu) [inline public/app/core/table_model.ts:45]:
```suggestion
    if (options.col == null || this.columns.length <= options.col) {
```

note that "== null" is the same as checking both null or undefined.  In general, we should prefer that over the `=== null|undefined`

Comment (hugohaggmark) [reply public/app/core/table_model.ts:45]: Why? I think the check for both `null` and `undefined` is much easier to read than the absence of one `=` is something you would easily miss during a review. Is this a new code style guideline that we're introducing and is it documented somewhere @ryantxu?

Comment (joshhunt) [reply public/app/core/table_model.ts:45]: I agree with Hugo. I strongly prefer the explicitness of Alexa's approach, which has less risks of someone who doesn't know the full history/context of Grafana coming in, misunderstanding the code, and reintroducing the bug.

Comment (ryantxu) [reply public/app/core/table_model.ts:45]: Fine by me --- we just need to be careful that treating null vs undefined differently is not going to work with anything coming from settings (that may be generated/processed by backend schemas and validation)

**Review by ryantxu — APPROVED** [review]

**Review by dprokop — APPROVED** [review]

**Pull request merged**